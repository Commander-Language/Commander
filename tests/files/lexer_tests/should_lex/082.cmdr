//Test that custom types lex correctly
type AddFunction = (int, int) -> int;
add: AddFunction = (a, b) -> a + b;