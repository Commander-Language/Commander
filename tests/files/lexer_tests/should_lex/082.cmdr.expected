TYPE 'type' 2:1
VARIABLE 'AddFunction' 2:6
EQUALS '=' 2:18
LPAREN '(' 2:20
INT 'int' 2:21
COMMA ',' 2:24
INT 'int' 2:26
RPAREN ')' 2:29
LAMBDA '->' 2:31
INT 'int' 2:34
SEMICOLON ';' 2:37
VARIABLE 'add' 3:1
COLON ':' 3:4
VARIABLE 'AddFunction' 3:6
EQUALS '=' 3:18
LPAREN '(' 3:20
VARIABLE 'a' 3:21
COMMA ',' 3:22
VARIABLE 'b' 3:24
RPAREN ')' 3:25
LAMBDA '->' 3:27
VARIABLE 'a' 3:30
ADD '+' 3:32
VARIABLE 'b' 3:34
SEMICOLON ';' 3:35
END_OF_FILE 3:36
